### PR TITLE
Web server merge request

### PR DESCRIPTION
**Pull request by Retailed**
I'm not really sure if README is good enough on "how to use" part. If you have more ideas, please feel free to contribute!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +19,26 @@ usages and source codes, we expect to distinguish modules, which can be centrali
 
 ## How to use
 
+While working on this project, we had to goals in mind: to help to detect important functions in Wikimedia projects
```
Comment (tanny411) [inline README.md]:
```suggestion
While working on this project, we had two goals in mind: to help detect important functions in Wikimedia projects 
```

File: README.md [hunk 2]
```diff
@@ -17,6 +19,26 @@ usages and source codes, we expect to distinguish modules, which can be centrali
 
 ## How to use
 
+While working on this project, we had to goals in mind: to help to detect important functions in Wikimedia projects 
+and to help to detect similar ones. This information can be viewed in [our site](https://abstract-wiki-ds.toolforge.org/).
```
Comment (tanny411) [inline README.md]:
```suggestion
and also detect similar ones. This information can be viewed in [our site](https://abstract-wiki-ds.toolforge.org/).
```

File: README.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # Abstract Wikipedia Data Science
 
-Repository for fetching and analyzing community functions across all wiki projects.
+Repository for fetching and analyzing community functions across all wiki projects. 
+
+Visit [this link](https://abstract-wiki-ds.toolforge.org/) to access the results!
 
 This project is Outreachy 21 task, more info can be found in [phabricator](https://phabricator.wikimedia.org/T263678).
```
Comment (tanny411) [inline README.md:7]:
```suggestion
This project is Outreachy 21 task, more info can be found in [phabricator](https://phabricator.wikimedia.org/T263678).

## Overview
Across various wiki projects (Wikipedia, Wikibooks, Wiktionary, etc) and across languages are numerous Lua functions, we call wikifunctions or modules, performing operations that reflect in templates or wiki pages. With the goal towards Abstract Wikipedia - a language-independent Wikipedia that generates wiki pages and articles of different languages from a pool of knowledge - it is now necessary to pool the community authored functions as well. This project gives users and contributors a place to analyze and start merging wikifunctions starting with *important* modules and then merging or refactoring *similar* modules.
```

File: README.md [hunk 4]
```diff
@@ -17,6 +19,26 @@ usages and source codes, we expect to distinguish modules, which can be centrali
 
 ## How to use
 
+While working on this project, we had to goals in mind: to help to detect important functions in Wikimedia projects 
+and to help to detect similar ones. This information can be viewed in [our site](https://abstract-wiki-ds.toolforge.org/).
+
+It is hard to determine, which parameters we consider important when working with Lua scripts. 
+That's the reason why the first step in working with the site is to set up weights for different script parameters,
```
Comment (tanny411) [inline README.md]:
```suggestion
That's the reason why the first step in working with the site is to set up weights for different parameters,
```

File: README.md [hunk 5]
```diff
@@ -17,6 +19,26 @@ usages and source codes, we expect to distinguish modules, which can be centrali
...
+such as amount of unique editors, amount of transclusions (links for this module on another wikis) and so on.
```
Comment (tanny411) [inline README.md]:
```suggestion
such as number of unique editors, number of edits, number of pages a module is transcluded in, and so on.
```

File: README.md [hunk 6]
```diff
@@ -17,6 +19,26 @@ usages and source codes, we expect to distinguish modules, which can be centrali
...
+This is done through the corresponding fields on the top of the page. Additionally, you can filter results you want 
+to get by using tabs in light blue box: filtering by Wikimedia project family (wiktionary, wikipedia, wikiversity etc.),
+filtering by the language the project uses and excluding modules, which only store data, but don't process it, 
+is available. By default, all Wikimedia projects and all languages are chosen and filtering for "data modules" is off.
+
+Clicking "request" sends the request to the server, which returns top 50 functions, corresponding to set filters 
+and features weights. The titles of these functions are clickable links, which lead to the corresponding pages 
+for viewing information on the single script. This page in its header contains module's title, name of the wiki,
```
Comment (tanny411) [inline README.md]:
```suggestion
and features weights. The titles of these functions are clickable links, which lead to the corresponding pages 
for viewing information of the those modules. This page in its header contains module's title, name of the wiki, 
```

File: README.md [hunk 7]
```diff
@@ -17,6 +19,26 @@ usages and source codes, we expect to distinguish modules, which can be centrali
 
 ## How to use
 
+While working on this project, we had to goals in mind: to help to detect important functions in Wikimedia projects 
+and to help to detect similar ones. This information can be viewed in [our site](https://abstract-wiki-ds.toolforge.org/).
+
+It is hard to determine, which parameters we consider important when working with Lua scripts. 
+That's the reason why the first step in working with the site is to set up weights for different script parameters,
+such as amount of unique editors, amount of transclusions (links for this module on another wikis) and so on.
+This is done through the corresponding fields on the top of the page. Additionally, you can filter results you want
```
Comment (tanny411) [inline README.md]:
```suggestion
This is done through the corresponding fields on the top of the page. The weights can be any values, just give higher values for features that you want more focus in. They are normalised anyways.

Additionally, you can filter results you want 
```

File: README.md [hunk 8]
```diff
@@ -17,6 +19,26 @@ usages and source codes, we expect to distinguish modules, which can be centrali
...
+This is done through the corresponding fields on the top of the page. Additionally, you can filter results you want 
+to get by using tabs in light blue box: filtering by Wikimedia project family (wiktionary, wikipedia, wikiversity etc.),
+filtering by the language the project uses and excluding modules, which only store data, but don't process it, 
+is available. By default, all Wikimedia projects and all languages are chosen and filtering for "data modules" is off.
+
+Clicking "request" sends the request to the server, which returns top 50 functions, corresponding to set filters 
+and features weights. The titles of these functions are clickable links, which lead to the corresponding pages 
+for viewing information on the single script. This page in its header contains module's title, name of the wiki, 
+where it was fetched from, and page ID in this wiki; grey box shows the source code of this module, 
+and on the right links to the functions, which are considered to be similar to current script, are displayed.
```
Comment (tanny411) [inline README.md]:
```suggestion
and on the right links to the functions, which are considered to be similar to current module, are displayed. 
```

File: README.md [hunk 9]
```diff
@@ -17,6 +19,26 @@ usages and source codes, we expect to distinguish modules, which can be centrali
...
+filtering by the language the project uses and excluding modules, which only store data, but don't process it, 
+is available. By default, all Wikimedia projects and all languages are chosen and filtering for "data modules" is off.
+
+Clicking "request" sends the request to the server, which returns top 50 functions, corresponding to set filters 
+and features weights. The titles of these functions are clickable links, which lead to the corresponding pages 
+for viewing information on the single script. This page in its header contains module's title, name of the wiki, 
+where it was fetched from, and page ID in this wiki; grey box shows the source code of this module, 
+and on the right links to the functions, which are considered to be similar to current script, are displayed. 
+
+This way the website allows working with both functions "importance" through setting up weights of scripts' parameters
```
Comment (tanny411) [inline README.md]:
```suggestion
This way the website allows working with both functions "importance" through setting up weights 
```

File: web/app.py [hunk 10]
```diff
@@ -33,55 +41,54 @@ def ping_pong():
     return jsonify('pong!')
```
Comment (tanny411) [inline web/app.py:41]: should this stay?

**Review by tanny411 — COMMENTED** [review]
That's some amazing work, thanks a lot for this! I've updated the readme a bit.
I think we can add a section at the end for things todo/ further improvements as such:
```
## Further improvements
- Improve clustering: Test code2vec or other similar code-based methods to create embeddings
- Add pageviews as a feature (Find a way to use page dumps. APIs were tested but take too long)
- Provide *diffs* among similar modules (shows users parts of code to modularize or merge)
```
Add things you want to mention about web part.

The how to use part seems well written. I think it's ready to merge into main?

File: web/app.py (hunk 10 above)
Comment (Retailed) [reply web/app.py:41]: I believe something like this (maybe with different, harder path, for example, `/backend-availability`) should stay - it makes easier to detect, if something happens, whether the error is caused by front-end or back-end

**Comment by Retailed** [comment]
> That's some amazing work, thanks a lot for this! I've updated the readme a bit.
> I think we can add a section at the end for things todo/ further improvements as such:
> 
> ```
> ## Further improvements
> - Improve clustering: Test code2vec or other similar code-based methods to create embeddings
> - Add pageviews as a feature (Find a way to use page dumps. APIs were tested but take too long)
> - Provide *diffs* among similar modules (shows users parts of code to modularize or merge)
> ```
> 
> Add things you want to mention about web part.
> 
> The how to use part seems well written. I think it's ready to merge into main?

Yes, I was thinking of merging this one with develop, then updating README in develop with further plans (and maybe adding some contact info, as e-mails?) and then merging with master!

File: web/app.py (hunk 10 above)
Comment (tanny411) [reply web/app.py:41]: Great!

**Pull request merged**